freshmaker/pyxis.py

```diff
@@ -0,0 +1,205 @@
+        ret.extend(response_data.get('data', []))
+        page = 1
+        last_page = ceil(response_data['total'] / response_data['page_size'])
+        while page < last_page:
+            params.update({"page": page})
+            response_data = self._make_request(entity, params=params)
+            ret.extend(response_data.get('data', []))
+            page += 1
+        return ret
+
+    def get_operator_indices(self, params={}):
+        """
+        Get all index images for organization(s)(configurable) from Pyxis
+
+        :param dict params: parameters to add to GET request
+        """
+        organization = conf.pyxis_index_image_organization
+        params.update({"filter": "organization=="+organization})
+
+        entity = "operators/indices"
+        indices = self._pagination(entity, params)
+        return indices
+
+    def _get_bundles_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = self._default_include_fields + \
+                         ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest']
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            params.update({"filter": f"source_index_container_path=={path}",
+                           "include": ','.join(include_fields)})
+            bundles = self._pagination('operators/bundles', params)
+            bundles_per_index_image.update({path: bundles})
+
+        return bundles_per_index_image
+
+    def get_bundles_per_channel_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per channel per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret = {}
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle.get('channel_name')
+                if channel in bundle_per_channel:
+                    new_version = bundle.get('version')
+                    old_version = bundle_per_channel[channel].get('version')
+                    if version.parse(new_version) > version.parse(old_version):
```

Comment:
Eventually, we will have a way to filter on if a bundle image uses semver mode for determining its upgrade graph. Until we can filter on that, you'll need to guard against the version not being a semantic version since this is not enforced unless the bundle uses `semver` mode.

Reply:
So I can suggest to add try-except block when parsing versions and if exception raised just skip this bundle for now. Will it be enough?

Reply:
@gnaponie This probably can help with that warning you mentioned and why it's there